accepted/install-locations.md

```diff
@@ -0,0 +1,227 @@
+# Install location search for .NET Core
+
+.NET Core has several cases where it needs to search for the install location of the runtime and SDKs. This document described the proposed algorithm for how this search will be done on all the supported platforms.
+
+## .NET Core install layout
+The content of a .NET Core install location always follows the same layout. The directory structure is the same on all platforms, file names may differ to conform with the platform specific requirements. (Below sample uses file names for Windows):
+* `./dotnet.exe` - the "muxer" which is used to execute apps and issue dotnet CLI commands.
+* `./host/fxr/<version>/hostfxr.dll` - this library contains all the logic to resolve framework references and select SDK. Regardless of application, the latest available version is used.
+* `./shared` - contains shared frameworks.
+  * `./Microsoft.NETCore.App/<version>` - the root framework which contains the .NET Core runtime. This framework contains `hostpolicy.dll` which is the entry point called by the `hostfxr` to run the app. Applications can choose which version to use.
+  * `./<FrameworkName>/<version>` - any other shared framework. Applications can choose which version to use.
+* `./sdk/<version>` - only installed when .NET Core SDK is installed - contains the SDK tools. Users can choose which version to use.
+
+
+## Installation types
+.NET Core can be installed in several ways, each provides a different location and expected behavior. Each such location will contain the .NET Core install layout described above.
+
+### Global install to default location
+This is the most common case where an installer installs the runtime (and potentially SDK) to a global default location accessible to everything on the system (and by all users).
+* Windows
+  * 32-bit on 64-bit OS - `%ProgramFiles(x86)%\dotnet`
+  * Otherwise - `%ProgramFiles%\dotnet`
+* Linux - the host searches in `/usr/share/dotnet` as a fallback, but the installer may pick a different default based on the requirements of the specific distro/configuration. In such case the install is effectively just like the global install to custom location described below. If correctly registered the host will search the custom location first.
+* macOS - `/usr/local/share/dotnet`
+
+*This type of install is supported by .NET Core 2.2 and earlier.*
```

Comment:
May be worth mentioning that finding this location can be done via the path, apphost, etc.

Reply:
For completeness, it may be worth calling out which installations modify the system path versus creating a symlink in order to find the global host location.

Reply:
I added description that we "register" the muxer - I don't want to go into the exact details as they are apparently different on platforms and install packaging solution.